cristin-project/src/main/java/no/unit/nva/cristin/projects/common/CristinQueryBuilder.java

```diff
@@ -0,0 +1,613 @@
+        cristinQuery.setValue(LEVELS, levels);
+        return this;
+    }
+
+    /**
+     * Setter only those projects that have been modified since this date will be returned.
+     */
+    public CristinQueryBuilder withModifiedSince(String modifiedSince) {
+        cristinQuery.setValue(PROJECT_MODIFIED_SINCE, modifiedSince);
+        return this;
+    }
+
+    /**
+     * Setter Name.
+     */
+    public CristinQueryBuilder withName(String name) {
+        cristinQuery.setValue(NAME, name);
+        return this;
+    }
+
+    /**
+     * Setter Organization.
+     */
+    public CristinQueryBuilder withOrganization(String organization) {
+        if (nonNull(organization)) {
+            cristinQuery.setValue(PROJECT_ORGANIZATION, organization);
+        } else {
+            cristinQuery.setValue(QUERY, EMPTY_STRING);
+        }
+        return this;
+    }
+
+    /**
+     * Setter a participant of the project by Cristin id, name or part of the name.
+     */
+    public CristinQueryBuilder withParentUnitId(String parentUnitId) {
+        if (nonNull(parentUnitId)) {
+            var unitId = parentUnitId;
+            if (parentUnitId.matches(PATTERN_IS_URL)) {
+                unitId = getUnitIdFromOrganization(parentUnitId);
+            }
+            cristinQuery.setPath(PATH_ORGANISATION, unitId);
+        }
+        return this;
+    }
+
+    /**
+     * Setter a participant of the project by Cristin id, name or part of the name.
+     */
+    public CristinQueryBuilder withParticipant(String participant) {
+        cristinQuery.setValue(PROJECT_PARTICIPANT, participant);
+        return this;
+    }
+
+    /**
+     * Setter Identity.
+     */
+    public CristinQueryBuilder withPathIdentity(String identity) {
+        System.out.println("withPathIdentity -> " + identity);
+        if (nonNull(identity) && !identity.isBlank()) {
+            if (identity.matches(PATH_ORGANISATION.getPattern())) {
+                cristinQuery.setPath(PATH_ORGANISATION, identity);
+            } else if (identity.matches(PATH_PROJECT.getPattern())) {
+                cristinQuery.setPath(PATH_PROJECT, identity);
+            } else {
+                cristinQuery.setPath(IDENTITY, identity);
+            }
+        } else {
+            cristinQuery.setPath(IDENTITY, EMPTY_STRING);
+        }
+        return this;
+    }
+
+    /**
+     * Setter Organization.
+     */
+    public CristinQueryBuilder withPathOrganization(String organization) {
+        System.out.println("withPathOrganization -> " + organization);
+
+        if (nonNull(organization)) {
+            if (organization.matches(PROJECT_ORGANIZATION.getPattern())) {
+                cristinQuery.setValue(PROJECT_ORGANIZATION,organization);
+            } else {
+                cristinQuery.setPath(PATH_ORGANISATION, organization);
+            }
+        } else {
+            cristinQuery.setValue(PATH_ORGANISATION, EMPTY_STRING);
```

Comment:
Is this needed?

Reply:
Yes, as it will will fail correctly when validating  (invalid value, instead of missing parameter)